hack/gen-token/main.go

```diff
@@ -0,0 +1,152 @@
+//go:build hack
+// +build hack
+
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ Usage:
+  go run ./hacks/gen-token/main.go --cert=<path>/<to>/ca.crt --url=https://<controller-ip>:<port>/
+
+*/
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+	"sigs.k8s.io/yaml"
+
+	"github.com/k0sproject/k0s/pkg/token"
+)
+
+var (
+	certPath string
+	joinURL  string
+	role     string
+	outDir   string
+	valid    time.Duration
+)
+
+func main() {
+	flag.StringVar(&certPath, "cert", "", "path to the certificate file")
+	flag.StringVar(&joinURL, "url", "", "url of the api server to join")
+	flag.StringVar(&role, "role", "worker", "token role. valid values: worker, controller. Default: worker")
+	flag.StringVar(&outDir, "out", ".", "path to the output directory. Default: current dir")
+	flag.DurationVar(&valid, "valid", time.Hour*24*365, "how long token is valid, in Go duration format. Default: 365 days")
```

Comment:
`k0s token create ...` default to non expiring tokens. maybe this should behave the same by default for consistency?

Reply:
Done